### PR TITLE
1570647 - Do not send 'metrics' ping if the app was not used

### PR DESCRIPTION
**Pull request by travis79**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt [hunk 1]
```diff
@@ -262,6 +275,32 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
...
+     * Update flag to show we are no longer in the foreground.
+     */
+    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
+    fun onEnterBackground() {
+        isInForeground = false
+    }
+
+    /**
+     * Update the flag to indicate we are moving to the foreground, and if Glean is initialized we
+     * will check to see if the metrics ping needs scheduled for collection.
```
Comment (badboy) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]:
```suggestion
     * will check to see if the metrics ping needs to be scheduled for collection.
```

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt [hunk 2]
```diff
@@ -31,8 +34,11 @@ import java.util.concurrent.TimeUnit as AndroidTimeUnit
  * - ping is overdue (due time already passed) for the current calendar day;
  * - ping is soon to be sent in the current calendar day;
  * - ping was already sent, and must be scheduled for the next calendar day.
+ *
+ * The scheduler also makes use of the [LifecycleObserver] in order to correctly schedule
```
Comment (badboy) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]: Do these btw get correctly linked?

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt [hunk 3]
```diff
@@ -31,8 +34,11 @@ import java.util.concurrent.TimeUnit as AndroidTimeUnit
...
+ * the [MetricsPingWorker]
```
Comment (badboy) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]:
```suggestion
 * the [MetricsPingWorker].
```

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt [hunk 4]
```diff
@@ -262,6 +276,32 @@ internal class MetricsPingScheduler(val applicationContext: Context) {
...
+    fun onEnterForeground() {
+        isInForeground = true
+
+        // We check for the metrics ping schedule here because the app could have been in the
+        // background and resumed in which case Glean would already be initialized but we still need
+        // to perform the check to determine whether or not to collect and schedule the metrics ping.
+        // If this is the first ON_START event since the app was launched, Glean wouldn't be
+        // initialized yet.
+        if (Glean.isInitialized()) {
+            Glean.metricsPingScheduler.schedule()
```
Comment (badboy) [inline glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]: Isn't this essentially `this.schedule()` or is there any difference referencing it through the `Glean.metricsPingScheduler`?

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt (hunk 2 above)
Comment (travis79) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]: They should, I've not checked in the glean-core repo but this does work in the docs that are generated in A-C

File: glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt (hunk 4 above)
Comment (travis79) [reply glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/MetricsPingScheduler.kt]: Yes, good point!  It is my fault for not changing this when I moved it from the `GleanLifecycleObserver`.

**Pull request merged**